rir/src/ir/Compiler.cpp

```diff
@@ -1073,6 +1073,87 @@ bool compileSpecialCall(Context& ctx, SEXP ast, SEXP fun, SEXP args_) {
     return false;
 }
 
+SEXP findRIRClosure(SEXP sym, SEXP rho) {
+    SEXP fun;
+    while (rho != R_EmptyEnv) {
+        fun = findVarInFrame3(rho, sym, TRUE);
+        if (fun != R_UnboundValue) {
+            if (TYPEOF(fun) == PROMSXP || TYPEOF(fun) == BUILTINSXP ||
+                    TYPEOF(fun) == SPECIALSXP || fun == R_MissingArg)
+                return nullptr;
+            if (isValidClosureSEXP(fun))
+                return fun;
+        }
+        rho = ENCLOS(rho);
+    }
+    return nullptr;
+}
+
+// try to look up a closure
+bool compileWithGuess(Context& ctx, SEXP ast, SEXP fun, SEXP args_) {
+    RList args(args_);
+    CodeStream& cs = ctx.cs();
+
+    SEXP cls = findRIRClosure(fun, R_GlobalEnv);  // or current env of the closure?
+    if (!cls)
+        return false;
+
+    RList formals(FORMALS(cls));
+    if (formals.length() != args.length() || args.length() == 0)
```

Comment:
Stupid question, but why do we check that the length of args is nonzero?

Reply:
No reason, just a leftover from debugging I guess.